src/cascade/input_data/configuration/construct_bundle.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import pandas as pd
+
+from cascade.input_data import InputDataError
+from cascade.input_data.configuration.construct_study import \
+    get_bundle_study_covariates
+from cascade.input_data.configuration.id_map import make_integrand_map
+from cascade.input_data.db.bundle import _get_bundle_id, _get_bundle_data
+from cascade.dismod.db.metadata import DensityEnum
+from cascade.core import getLoggers
+
+CODELOG, MATHLOG = getLoggers(__name__)
+
+
+def _normalize_measures(data):
+    """Transform measure_ids into canonical measure names, for instance,
+    GBD measure 38 for birth prevalence becomes pini.
+    """
+    data = data.copy()
+    gbd_measure_id_to_integrand = make_integrand_map()
+    if any(data.measure_id == 6):
+        MATHLOG.warn(f"Found incidence, measure_id=6, in data. Should be Tincidence or Sincidence.")
+    try:
+        data["measure"] = data.measure_id.apply(lambda k: gbd_measure_id_to_integrand[k].name)
+    except KeyError as ke:
+        raise RuntimeError(
+            f"The bundle data uses measure {str(ke)} which doesn't map "
+            f"to an integrand. The map is {gbd_measure_id_to_integrand}."
+        )
+    return data
+
+
+def _normalize_sex(data):
+    """Transform sex_ids from 1, 2, 3 to male, female, both.
+    """
+    data = data.copy()
+    try:
+        data["sex"] = data.sex_id.apply({1: "Male", 2: "Female", 3: "Both", 4: "Unspecified"}.get)
+    except KeyError as ke:
+        raise InputDataError(f"Unrecognized sex id") from ke
+    return data
+
+
+def _normalize_bundle_data(data):
+    """Normalize bundle columns, strip extra columns and index on `seq`
+    """
+    data = _normalize_measures(data)
+    data = _normalize_sex(data)
+
+    data = data.set_index("seq")
+
+    cols = ["measure", "mean", "sex", "sex_id", "standard_error", "age_start", "age_end", "year_start", "year_end", "location_id"]
```

Comment:
This line is long.  I can't see it all in the browser.  I don't see a scroll bar.

Reply:
haha